### DISCUSSION TITLE
Code not suited for strict compile rules

### DISCUSSION THREAD

**Discussion by lefebvresam** [Azure/azure-iot-sdk-c#1981] [Ideas]
**Development Machine, OS, Compiler (and Other Relevant Toolchain Info)**

WSL2 on Windows10 education pro
Cross compiled for arm on Ubuntu 20.04

**SDK Version (Please Give Commit SHA if Manually Compiling)**

1.4.1

**Describe the Bug**

When compiling with strict compile rules as already available in an existing project, there are a lot of 'pendantic' and other errors. To get rid of them you need to compile without -Wall, -pendantic and -Werror. This makes it difficult to mix compilation with other libs.

```
  target_compile_options(${PROJECT_NAME}
                         PUBLIC -Wall
                                -Wextra
                                -pedantic
                                -Wno-implicit-fallthrough
                                -Wno-class-memaccess
                                -D${ARCH}
                                ${PARSED_ARGS_EXTRA_PUBLIC_COMPILE_OPTIONS}
                         PRIVATE -Werror ${PARSED_ARGS_EXTRA_PRIVATE_COMPILE_OPTIONS})
```

**Comment by momuno** [comment]
Hi @lefebvresam , thanks for bringing this to our attention.  We will be looking into it.
